api/v1alpha1/port_types.go

```diff
@@ -122,6 +122,23 @@ type PortResourceSpec struct {
 	// +listType=set
 	// +optional
 	SecurityGroupRefs []OpenStackName `json:"securityGroupRefs,omitempty"`
+
+	// vnicType specifies the type of vNIC which this port should be
+	// attached to. This is used to determine which mechanism driver(s) to
+	// be used to bind the port. The valid values are normal, macvtap,
+	// direct, baremetal, direct-physical, virtio-forwarder, smart-nic and
+	// remote-managed, although these values will not be validated in this
+	// API to ensure compatibility with future neutron changes or custom
+	// implementations. What type of vNIC is actually available depends on
+	// deployments. If not specified, the Neutron default value is used.
+	// +kubebuilder:validation:MaxLength:=1024
```

Comment:
The size of the `vnic_type` field in neutron DB is 64, can you update the MaxLength validation to this value?

https://github.com/openstack/neutron/blob/e47aed1508badcf2f52e0a5ced932abf9ff73655/neutron/plugins/ml2/models.py#L44